### DISCUSSION TITLE
SCNetworkReachability API deprecations.

### DISCUSSION THREAD

**Discussion by FONickReichard** [firebase/firebase-ios-sdk#12592] [General]
Today, I discovered that starting with iOS 17.4, several methods within the SCNetworkReachability API collection are being marked as deprecated.
https://developer.apple.com/documentation/systemconfiguration/scnetworkreachability-g7d

It looks like the [FirebasePerformance/Sources/FPRNanoPbUtils.m](https://github.com/firebase/firebase-ios-sdk/blob/351230124e2046b2f78db1ff1c0d472885f46835/FirebasePerformance/Sources/FPRNanoPbUtils.m#L84) currently uses `SCNetworkReachabilityGetFlags`, which is apart of the deprecations.

I wanted to pass along this deprecation notice, as it seems to be new, and ensure that y'all can catch it early if it's not already on the road map.

Best, 

Nick 

PS: Thanks for all the goods you create.

**Comment by paulb777** [comment]
Thanks @FONickReichard! I'll convert to an issue. It looks like something for both Performance and Firestore to investigate:

```
 $ git grep SCNetworkReachabilityGetFlags
FirebasePerformance/Sources/FPRNanoPbUtils.m:  SCNetworkReachabilityGetFlags(reachabilityRef, &reachabilityFlags);
Firestore/core/src/remote/connectivity_monitor_apple.mm:    if (SCNetworkReachabilityGetFlags(reachability_, &flags)) {
Firestore/core/src/remote/connectivity_monitor_apple.mm:    if (!SCNetworkReachabilityGetFlags(reachability_, &flags)) return;
```
